executor/hnswpsql.py

```diff
@@ -3,7 +3,11 @@
 
 import copy
 import inspect
-from datetime import datetime
+import threading
+import time
+import traceback
```

Comment:
is this import needed?

Reply:
Yes, so we can get a stack trace when the thread fails.

Reply:
Maybe we need to subclass Thread as we did once in the core to better handle exceptions. That should be another PR. This one is big enough as is.